deploy/modules/gcp/tidb-operator/main.tf

```diff
@@ -53,24 +53,13 @@ resource "null_resource" "get-credentials" {
   }
 }
 
-data "local_file" "kubeconfig" {
-  depends_on = [google_container_cluster.cluster, null_resource.get-credentials]
-  filename   = var.kubeconfig_path
-}
-
-resource "local_file" "kubeconfig" {
-  depends_on = [google_container_cluster.cluster, null_resource.get-credentials]
-  content    = data.local_file.kubeconfig.content
-  filename   = var.kubeconfig_path
-}
```

Comment:
The `helm_release.tidb-operator` is applied after resource `null_resource.get-credentials`. I don't think we need this resource, it is `null_resource.get-credentials` to create the file.

https://github.com/pingcap/tidb-operator/blob/d22166c4ef8924f5b8b50348aa25c801da76b4d8/deploy/modules/gcp/tidb-operator/main.tf#L125-L131

Reply:
One known issue is when I accidentally deleted the kubeconfig file, `terraform apply` will not run `null_resource.get-credentials` again. local_file data/resource cannot resolve this issue too. I guess we need to find some way to trigger `null_resource.get-credentials` to rerun when file does not exist.

Current workaround: `terraform apply -target module.tidb-operator.null_resource.get-credentials`

Reply:
Would it be better to do something like what is described in this issue? https://github.com/terraform-providers/terraform-provider-google/issues/3097

Reply:
it seems better

Reply:
I'm wrong here, there is no way to create the file in refresh state, because [terraform refresh](https://www.terraform.io/docs/commands/refresh.html) does not modify infrastructure.

Reply:
however, we can try to make it better. for now, `terraform state rm module.tidb-operator.null_resource.get-credentials` is required, otherwise `terraform apply -target module.tidb-operator.null_resource.get-credentials` will not create the file because the file creation is a side effect not the desired state of the resource